sw-precache.js

```diff
@@ -119,11 +120,13 @@ module.exports = function(params) {
   params.logger(util.format('Total precache size is about %s for %d resources.',
     formatBytesAsString(cumulativeSize), relativeUrls.length));
 
-  var templateBuffer = fs.readFileSync(params.templateFilePath);
-  return _.template(templateBuffer, {
+  var template = fs.readFileSync(params.templateFilePath, {encoding: 'utf8'});
+  var populatedTemplate = _.template(template, {
     handleFetch: params.handleFetch,
     importScripts: params.importScripts ? params.importScripts.map(JSON.stringify).join(',') : null,
     includeCachePolyfill: params.includeCachePolyfill,
     precacheConfig: JSON.stringify(precacheConfig)
   });
+
+  return resumer().queue(populatedTemplate).end();
```

Comment:
I'm not sure you're trying to force this to be stream when it get's no benefit from being it.

You can easily just use a callback based API in your gulpfile. There's no requirement that everything has to be streams.